fix: Avatar group - do not overlay outer shadow with inner component's shadow [ci visual]

N1XUS opened the pull request:
## Related Issue
Closes https://github.com/SAP/fundamental-styles/issues/2153

## Description
Add `overflow: hidden` for inner elements of `fd-popover__body` which will prevent inner component's shadow to overlay outer one.

## Screenshots
> **NOTE:** If you've made any style changes, please provide appropriate screenshots (before and after) to help reviewers.
>
> To see examples of which screenshots to include, go to [Screenshot Examples](https://github.com/SAP/fundamental-styles/wiki/Pull-Request-Screenshot-Examples).

### Before:
![image](https://user-images.githubusercontent.com/6586561/116597205-fd1b9d80-a92d-11eb-8fc2-293b25ff6540.png)


### After:
![image](https://user-images.githubusercontent.com/6586561/116597226-03aa1500-a92e-11eb-92cc-6e42e061c733.png)

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
- [x] last commit message should have `[ci visual]` so it can trigger chromatic visual regression (e.g. `test: run chromatic visual regression [ci visual]`)
4. Documentation
- [x] Storybook documentation has been created/updated
- [x] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

InnaAtanasova commented on src/popover.scss:
```diff
@@ -102,6 +102,11 @@ $block: #{$fd-namespace}-popover;
     }
 
+    // Inner shadows can affect outer shadow. This will discard such behavior
+    > * {
+      overflow: hidden;
+    }
```
We can't use this solution unfortunately. We have cases where the size of the popover body is limited and the scroll should appear, but if we add this solution the scroll no longer works. Please check "Scrollable" example of Popover. https://deploy-preview-2296--fundamental-styles.netlify.app/?path=/docs/components-popover--alignment#scrollable

The scroll is not working, if you remove the `overflow: hidden;` it will appear.

N1XUS commented:
@InnaAtanasova I've updated the selector also checked with all examples from popover and corrected one demo component in the documentation. Now should be fine. Please re-check.
